block/src/raw_async.rs

```diff
@@ -52,10 +52,23 @@ impl DiskFile for RawFileDisk {
     }
 }
 
+struct AioRequest {
+    offset: libc::off_t,
+    iovecs: Vec<libc::iovec>,
+    user_data: u64,
+    req_type: RequestType,
+}
+
+// SAFETY: data structure only contain a series of integers
+unsafe impl Send for RawFileAsync {}
+// SAFETY: data structure only contain a series of integers
+unsafe impl Sync for RawFileAsync {}
+
```

Comment:
I don't think the comments here are correct - the problem is about justifying `*mut c_void` is thread safe, which I don't think it is.

Reply:
If you see the code of allocating SmallVec, it's inside a for loop eventually and short lived. I might not have enough expertise how to propagate the reference to the actual cached vector. If we could do that might an improvement over copying the vector using to_vec.